### PR TITLE
db refine

### PR DESCRIPTION
**Pull request by fluiderson**

### CODE REVIEW

**Comment by Vovke** [comment]
by doing this "folder(chain2) based code backup strategy" you will eventually override the commit history of the code. Please ensure that it's not happening.

File: src/signer.rs [hunk 1]
```diff
@@ -1,157 +1,249 @@
...
+        &self,
+        id: impl AsRef<[u8]>,
+    ) -> Result<AccountId32, SignerError> {
+        Pair::from_entropy_and_full_derivation(&self.pair.1 .0, derive_invoice_junction(id))
+            .map(|pair| AccountId32(pair.public().0))
+            .map_err(Into::into)
+    }
+
+    pub fn sign(&self, id: impl AsRef<[u8]>, msg: &[u8]) -> Result<Signature, SignerError> {
+        let mut random = rand::thread_rng();
```
Comment (Slesarew) [inline src/signer.rs:234]: seriously? This is exactly why you should not be implementing any of this manually. This is supposed to internally call OsRnd which, by the way, IS blocking, thus threading here.

Comment (fluiderson) [reply src/signer.rs:234]: Do you mean that `rand::thread_rng()` does block for too long, or?.. If you do, then I don't quite understand for what we need to save so much time. And by the way, we already use the same function [here](https://github.com/paritytech/soketto/blob/cd261c9bf22e4177cc70deddd1c8aabc40715a51/src/handshake/client.rs#L133) on every payout (as we create a new connection each time).

**Comment by fluiderson** [comment]
Moved to https://github.com/Alzymologist/Kalatori-backend/pull/71.

**Pull request closed without merging**